tests/validator.go

```diff
@@ -139,6 +141,49 @@ func formatUUID(s string) string {
 	return strings.ToUpper(strings.Replace(s, "-", "", -1))
 }
 
+func validateClevisBinding(t *testing.T, expected []*types.Partition) error {
+	for _, e := range expected {
+		if e.ClevisBinding != "" && e.LuksDeviceName == "" {
+			return fmt.Errorf("Expected LuksDeviceName for ClevisBinding %s", e.ClevisBinding)
+		}
+
+		switch e.ClevisBinding {
+		case "":
+			continue
+		case "tpm2":
+			output, err := getLuksDump(e.LuksDeviceName)
+			if err != nil {
+				return fmt.Errorf("Error getting luks metadata: %v", err)
+			}
+			if len(output.Config.Flags) > 0 && output.Config.Flags[0] != "tpm2" {
+				return fmt.Errorf("Expected tpm2 binding, got %s", output.Config.Flags[0])
+			}
+			continue
+		default:
+			return fmt.Errorf("Unknown clevis binding: %s", e.ClevisBinding)
+		}
+	}
+	return nil
+}
+
+type LuksDump struct {
+	Config struct {
+		Flags []string `json:"flags"`
+	} `json:"config"`
+}
+
+func getLuksDump(devAlias string) (LuksDump, error) {
+	dump, err := exec.Command("sudo", distro.CryptsetupCmd(), "luksDump", "--dump-json-metadata", devAlias).CombinedOutput()
```

Comment:
Ugh, in concept this should work, but for a reason or another the device no longer exist by the time validation occurs even though I am defering clean up till after this happens.  Any ideas?

Reply:
If I cannot figure the binding verification out before the end of sprint. I might spin off a issue for it and drop my prior commit so we can land some blackbox luks test value wdyt?

Reply:
Yeah, fine to skip the Clevis validation bits for now!